### PR TITLE
Slides For Session 7

### PR DESCRIPTION
**Pull request by JamieB-gu**
Added some slides for session 7.

### CODE REVIEW

File: 7-types/slides.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+# Types
+
+A more detailed look at types, and how TypeScript's approach to types compares to other languages. Topics include:
+
+- Types vs values
+- Structural typing vs nominal typing
+
+There's more information on how TypeScript handles structural typing in the TypeScript handbook: https://www.typescriptlang.org/docs/handbook/type-compatibility.html
```
Comment (mxdvl) [inline 7-types/slides.md:8]:
```suggestion
There's more information on how TypeScript handles structural typing [in the TypeScript handbook](https://www.typescriptlang.org/docs/handbook/type-compatibility.html)
```

File: 7-types/slides.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+  val bar: Bar = Bar()
+}
+```
+
+---
+
+## Classes In TypeScript
+
+As mentioned in session 5, TypeScript classes also create both types and values when they're defined:
```
Comment (mxdvl) [inline 7-types/slides.md:74]: Should we add a link to session 5?

**Review by mxdvl — APPROVED** [review]
LGTM!

We should probably mention in the session that the moment you create a private member in a class, [you effectively achieve Nominal typing in TypeScript](https://www.typescriptlang.org/play?#code/MYGwhgzhAEByYFsCmATaBvAUNH0DEAdokgFzQQAuATgJYEDmA3NrsAPYGVUCuwFbVABRFkZLnXoBKDC1y4KACxoQAdIWLQAvNBFJmcgL6ZZ0ekgo7ig6emhVz3KgWiLla3dCNHMoSDAAKSFQQHDJy6qLk1BL6uABWbABGYtEMsTjsnNS8-EK6KbQMADTQCclRhVJhcvJKqhFIWpbI6TWuqmVNZa3ecmYWutYYdg5OLnXuGl7GmZTQwAAOAMrsFBbaBEgA7tCBwRyCAOQAwv7QK2xrhyWHAFJsjkQgyhTXksw+HHMKSCAgbE1hMQUGR4MgUJIChItAA+EwAAwAEr9-tAACToXQoFS6Azwj6zCw-P5sU4XNZNYn-QSLckUd5AA)! It’s then possible to make it behave like Scala’s `trait` [with abstract classes](https://www.typescriptlang.org/play?#code/IYIwzgLgTsDGEAJYBthjAgcsAtgUwBMEBvAKAQoQGIA7XPALgUigEsaBzAbnMtgHsaLAK7x+UABR18TFuw4BKEr0qUIAC1ZgAdLXoIAvAml4eqgL6kVCDnkQmJS4gih3hUGgg1bdJhJctSFDQMAAU8KDBBBDwADwg8GgIMbHwiMlUAK34QWWh5Mz5BETFJEzy2TgAaBGzc5nzOJ2tVMGEABwipegVC1QpvHTrDWpy+-1JAgSFEWHaAZQEIRCMaPAB3BHDIwQkAcgBhUIRF-mW9mr2AKX53OmQtCD3eq2nIBHU8ZGR+Ee60pipQgKCryQwAPmsAAMABJfH4IAAkxBMBG0JnMUJ4QWKiE+334R1OyxG+J+EjmxIgvSAA).

File: 7-types/slides.md (hunk 2 above)
Comment (JamieB-gu) [reply 7-types/slides.md]: 👍 268ecbf

File: 7-types/slides.md (hunk 1 above)
Comment (JamieB-gu) [reply 7-types/slides.md:8]: In this case I've intentionally left the link unhidden so people can see it when the slides are being presented.

**Comment by JamieB-gu** [comment]
> We should probably mention in the session that the moment you create a private member in a class, you effectively achieve Nominal typing in TypeScript! It’s then possible to make it behave like Scala’s `trait` with abstract classes.

Created #41 to track this.

**Pull request merged**